File: app/src/main/java/ru/zulvit/SignUpFragment.java
package ru.zulvit;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import ru.zulvit.databinding.FragmentSignUpBinding;
import ru.zulvit.models.User;

public class SignUpFragment extends Fragment {
    private FragmentSignUpBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentSignUpBinding.inflate(inflater, container, false);

        binding.buttonSignUp.setOnClickListener(v -> {
            String username = binding.editTextUsername.getText().toString().trim();
            String email = binding.editTextEmail.getText().toString().trim();
            String password = binding.editTextPassword.getText().toString().trim();

            User user = new User(username, email, password);
            NavController navController = Navigation.findNavController(v);
            navController.navigate(SignUpFragmentDirections.actionSignUpFragmentToSignInFragment(user));
        });

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/ru/zulvit/ui/signUp/SignUpFragment.kt
package ru.zulvit.ui.signUp

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.Navigation
import ru.zulvit.data.models.User
import ru.zulvit.databinding.FragmentSignUpBinding

class SignUpFragment : Fragment() {

    private var _binding: FragmentSignUpBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentSignUpBinding.inflate(inflater, container, false)

        binding.buttonSignUp.setOnClickListener {
            val email = binding.editTextEmail.text.toString().trim()
            val password = binding.editTextPassword.text.toString().trim()

            // Создаем объект пользователя
            val user = User(email, "", password)

            // Передаем пользователя в SignInFragment
            val action = SignUpFragmentDirections.actionSignUpFragmentToSignInFragment(user)
            val navController: NavController = Navigation.findNavController(it)
            navController.navigate(action)
        }

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
